packages/paste-core/components/combobox/src/styles/ComboboxListboxOption.tsx

```diff
@@ -1,74 +1,103 @@
   },
 };
 
+const getVirtualStyles = (startHeight: number): Record<string, unknown> => ({
+  position: 'absolute' as PositionOptions,
+  top: 0,
+  left: 0,
+  width: '100%',
+  transform: `translateY(${startHeight}px)`,
+});
+
 const ComboboxListboxOption = React.forwardRef<HTMLLIElement, ComboboxListboxOptionProps>(
-  ({children, element = 'COMBOBOX', highlighted, variant = 'default', startHeight, ...props}, ref) => {
-    const virtualItemStyles =
-      startHeight !== undefined
-        ? {
-            position: 'absolute' as PositionOptions,
-            top: 0,
-            left: 0,
-            width: '100%',
-            transform: `translateY(${startHeight}px)`,
-          }
-        : {};
+  (
+    {
+      children,
+      element = 'COMBOBOX',
+      highlighted,
+      selected,
+      disabled = false,
+      variant = 'default',
+      startHeight,
+      ...props
+    },
+    ref
+  ) => {
+    const virtualItemStyles = startHeight != null ? getVirtualStyles(startHeight) : {};
+
     return (
       <Box
         {...safelySpreadBoxProps(props)}
         as="li"
+        disabled={disabled}
         element={`${element}_LIST_ITEM`}
-        backgroundColor={highlighted ? 'colorBackgroundPrimaryWeakest' : 'colorBackgroundBody'}
-        padding="space30"
-        cursor="pointer"
+        backgroundColor={highlighted && !disabled ? 'colorBackgroundPrimaryWeakest' : 'colorBackgroundBody'}
+        color={highlighted ? 'colorTextLinkStronger' : selected ? 'colorTextLink' : 'colorText'}
+        opacity={disabled ? 0.6 : 1}
+        display="flex"
+        justifyContent="space-between"
+        alignItems="center"
+        paddingY="space40"
+        cursor={disabled ? 'not-allowed' : 'pointer'}
         ref={ref}
+        transition="background-color 150ms ease, color 150ms ease"
+        borderBottomWidth="borderWidth10"
+        borderBottomColor="colorBorderDecorative10Weaker"
+        borderBottomStyle="solid"
+        _before={{
+          content: `""`,
+          position: 'absolute',
+          top: 0,
+          bottom: 0,
+          left: 0,
+          borderLeftWidth: 'borderWidth30',
+          borderLeftColor: highlighted && !disabled ? 'colorBorderPrimaryStronger' : 'transparent',
+          borderLeftStyle: 'solid',
+        }}
+        _last={{
+          marginBottom: 'space0',
+          borderBottomColor: 'transparent',
+        }}
         {...virtualItemStyles}
         {...VariantStyles[variant]}
       >
-        <Text
-          as="span"
-          color="colorText"
-          element={`${element}_LIST_ITEM_TEXT`}
-          textDecoration={highlighted ? 'underline' : null}
-        >
-          {children}
-        </Text>
+        <Box as="span">
+          <Text as="span" color="inherit" element={`${element}_LIST_ITEM_TEXT`}>
+            {children}
+          </Text>
+        </Box>
+        <Box as="span" hidden={selected ? false : true}>
+          <CheckboxCheckIcon decorative color="inherit" />
```

Comment:
issue: we probably need something to represent this icon to AT users. When using this with a screen reader, I actually get no indication that the item is part of a collection of selected items from the overall combobox. It might be a limitation with Downshift but as you arrow through the list, Downshift sets aria-selected to true for the highlighted index. So every items says "selected". This isn't necessarily "wrong" either, for the purposes of that interaction, moving through the list of options, it is the selected item in that list for the purpose of navigation.

Reply:
@shleewhite this is my commentary on your question

> Currently, the active item in the dropdown automatically has aria-selected="true". This makes it really confusing as a screen reader user because the item isnt actually selected until I press enter and because as I navigate through the list, every single item is announced as selected. The aria-selected attribute should instead be set depending on if the item is in the pill group.

Reply:
Fixed.